### ISSUE TITLE
Reactive Props Destructure supported in Vue 3.5 still not reactive in Nuxt 3.13.1 

### ISSUE THREAD

**Issue by santiagoaloi** [nuxt/nuxt#28885]
### Environment

```bash

------------------------------
- Operating System: Darwin
- Node Version:     v20.17.0
- Nuxt Version:     3.13.1
- CLI Version:      3.13.1
- Nitro Version:    2.9.7
- Package Manager:  pnpm@9.9.0
- Builder:          -
- User Config:      -
- Runtime Modules:  -
- Build Modules:    -
------------------------------
```

### Reproduction

Changed `showThemeName` from the parent and its not reactive in the child prop.
```js
<script setup lang="ts">
const { showThemeName = false } = defineProps<{
  /**
   * Whether to show the theme name.
   * @default false
   */
  showThemeName?: boolean
}>()
</script>
```

### Describe the bug

I defined `showThemeName` as a prop following the Reactive Props Destructure
approach supported in Vue 3.5, but still no reactivity.  

```js
<script setup lang="ts">
const { showThemeName = false } = defineProps<{
  /**
   * Whether to show the theme name.
   * @default false
   */
  showThemeName?: boolean
}>()
</script>
```

**Comment by santiagoaloi** [comment]
Could it be that its not released yet? #28830

**Comment by danielroe** [comment]
this is enabled by default in next nuxt release - for now, see https://github.com/nuxt/nuxt/issues/28829#issuecomment-2329411615.
